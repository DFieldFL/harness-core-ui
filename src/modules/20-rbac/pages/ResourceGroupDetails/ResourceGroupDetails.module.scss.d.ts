/* eslint-disable */
// this is an auto-generated file
declare const styles: {
  readonly headerContainer: string
  readonly resourceTypeListContainer: string
  readonly pageContainer: string
  readonly colorDiv: string
}
export default styles
