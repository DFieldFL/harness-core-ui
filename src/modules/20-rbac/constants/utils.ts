export enum RbacResourceGroupTypes {
  DYNAMIC_RESOURCE_SELECTOR = 'DynamicResourceSelector'
}
