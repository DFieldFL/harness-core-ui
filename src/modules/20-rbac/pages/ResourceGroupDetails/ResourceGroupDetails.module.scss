.headerContainer {
  background-color: var(--primary-1) !important;
}
.resourceTypeListContainer {
  background-color: var(--white) !important;
}
.pageContainer {
  display: grid;
  grid-auto-flow: column;
  grid-template-columns: 1fr 4fr;
  height: 100%;
}
.colorDiv {
  min-height: 50px;
  min-width: 50px;
}
