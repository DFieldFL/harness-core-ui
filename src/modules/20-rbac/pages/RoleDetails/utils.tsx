import produce from 'immer'
import { PermissionIdentifier } from '@rbac/interfaces/PermissionIdentifier'
import type { ResourceType } from '@common/interfaces/ResourceInterface'
import type { Permission, PermissionResponse } from 'services/rbac'

export const getPermissionMap = (permissionList?: PermissionResponse[]): Map<ResourceType, Permission[]> => {
  const permissionsMap: Map<ResourceType, Permission[]> = new Map()
  permissionList?.map(response => {
    if (response.permission.resourceType) {
      if (permissionsMap.has(response.permission.resourceType as ResourceType)) {
        permissionsMap.get(response.permission.resourceType as ResourceType)?.push(response.permission)
      } else permissionsMap.set(response.permission.resourceType as ResourceType, [response.permission])
    }
  })
  return permissionsMap
}

export const onPermissionChange = (
  permission: string,
  isAdd: boolean,
  permissions: string[],
  setPermissions: (value: React.SetStateAction<string[]>) => void
): void => {
  if (isAdd) {
    if (
      permission === PermissionIdentifier.EDIT_DASHBOARD ||
      permissions.indexOf(PermissionIdentifier.EDIT_DASHBOARD) !== -1
    ) {
      setPermissions(_permissions => [...permissions, permission, PermissionIdentifier.VIEW_DASHBOARD])
    } else {
      setPermissions(_permissions => [...permissions, permission])
    }
  } else if (
    !(
      permission === PermissionIdentifier.VIEW_DASHBOARD &&
      permissions.indexOf(PermissionIdentifier.EDIT_DASHBOARD) !== -1
    )
  ) {
    setPermissions(_permissions =>
      produce(_permissions, draft => {
        draft?.splice(permissions.indexOf(permission), 1)
      })
    )
  }
}
