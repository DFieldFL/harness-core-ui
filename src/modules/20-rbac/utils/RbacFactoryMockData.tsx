import React from 'react'
import type { ResourceHandler, ResourceTypeGroupHandler } from '@rbac/factories/RbacFactory'
import { ResourceType, ResourceTypeGroup } from '@rbac/interfaces/ResourceType'

export const getResourceTypeHandlerMock = (resource: ResourceType): ResourceHandler | undefined => {
  switch (resource) {
    case ResourceType.PROJECT:
      return {
        icon: 'nav-project',
        label: 'Projects',
        // eslint-disable-next-line react/display-name
        addResourceModalBody: () => <></>
      }
    case ResourceType.ORGANIZATION:
      return {
        icon: 'nav-project',
        label: 'Organizations',
        // eslint-disable-next-line react/display-name
        addResourceModalBody: () => <></>
      }
    case ResourceType.SECRET:
      return {
        icon: 'lock',
        label: 'Secrets',
        category: ResourceTypeGroup.PROJECT_RESOURCES,
        // eslint-disable-next-line react/display-name
        addResourceModalBody: () => <></>
      }
  }
}

export const getResourceGroupTypeHandlerMock = (
  resource: ResourceType | ResourceTypeGroup
): ResourceTypeGroupHandler | undefined => {
  switch (resource) {
    case ResourceTypeGroup.PROJECT_RESOURCES:
      return {
        icon: 'nav-project',
        label: 'Project Resources',
        resourceTypes: new Set([ResourceType.SECRET])
      }
    case ResourceTypeGroup.ADMINSTRATIVE_FUNCTIONS:
      return {
        icon: 'nav-project',
        label: 'Administrative Fucntions',
        resourceTypes: new Set([ResourceType.ORGANIZATION, ResourceType.PROJECT])
      }
  }
}
