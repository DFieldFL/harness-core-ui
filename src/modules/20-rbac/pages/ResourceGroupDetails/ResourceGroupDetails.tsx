import React, { useEffect, useState } from 'react'
import { Text, Layout, Container, Button, Color } from '@wings-software/uicore'
import { useParams } from 'react-router-dom'
import ReactTimeago from 'react-timeago'
import produce from 'immer'
import type { ResourceGroupDetailsPathProps } from '@common/interfaces/RouteInterfaces'

import { useStrings } from 'framework/strings'
import ResourceTypeList from '@rbac/components/ResourceTypeList/ResourceTypeList'
import {
  useGetResourceGroup,
  useUpdateResourceGroup,
  ResourceGroupRequestRequestBody,
  ResourceGroupDTO,
  useGetResourceTypes
} from 'services/platform'
import { Page } from '@common/components/Page/Page'
import { useToaster } from '@common/components/Toaster/useToaster'
import { RbacResourceGroupTypes } from '@rbac/constants/utils'
import type { ResourceType, ResourceCategory } from '@rbac/interfaces/ResourceType'
import ResourcesCard from '@rbac/components/ResourcesCard/ResourcesCard'
import { Breadcrumbs } from '@common/components/Breadcrumbs/Breadcrumbs'
import routes from '@common/RouteDefinitions'
import RbacFactory from '@rbac/factories/RbacFactory'
import { getResourceSelectorsfromMap, getSelectedResourcesMap } from './utils'
import css from './ResourceGroupDetails.module.scss'

const ResourceGroupDetails: React.FC = () => {
  const { accountId, projectIdentifier, orgIdentifier, resourceGroupIdentifier } = useParams<
    ResourceGroupDetailsPathProps
  >()
  const { getString } = useStrings()
  const { showError, showSuccess } = useToaster()
  const [isUpdated, setIsUpdated] = useState<boolean>(false)

  const [selectedResourcesMap, setSelectedResourceMap] = useState<Map<ResourceType, string[] | string>>(new Map())
  const [resourceCategoryMap, setResourceCategoryMap] = useState<
    Map<ResourceType | ResourceCategory, ResourceType[] | undefined>
  >()

  const { data: resourceGroupDetails, error: errorInGettingResourceGroup, loading, refetch } = useGetResourceGroup({
    identifier: resourceGroupIdentifier,
    queryParams: {
      accountIdentifier: accountId,
      projectIdentifier,
      orgIdentifier
    }
  })

  const { data: resourceTypeData } = useGetResourceTypes({
    queryParams: {
      accountIdentifier: accountId,
      projectIdentifier,
      orgIdentifier
    }
  })

  useEffect(() => {
    setResourceCategoryMap(
      RbacFactory.getResourceCategoryList(
        (resourceTypeData?.data?.resourceTypes?.map(val => val.name) || []) as ResourceType[]
      )
    )
  }, [resourceTypeData?.data])

  useEffect(() => {
    setSelectedResourceMap(getSelectedResourcesMap(resourceGroupDetails?.data?.resourceGroup.resourceSelectors))
    setIsUpdated(false)
  }, [resourceGroupDetails?.data?.resourceGroup])

  const { mutate: updateResourceGroup, loading: updating } = useUpdateResourceGroup({
    identifier: resourceGroupIdentifier || '',
    queryParams: {
      accountIdentifier: accountId,
      orgIdentifier,
      projectIdentifier
    }
  })

  const updateResourceGroupData = async (resourceGroup: ResourceGroupDTO): Promise<void> => {
    const dataToSubmit: ResourceGroupRequestRequestBody = {
      resourcegroup: {
        ...resourceGroup,
        resourceSelectors: getResourceSelectorsfromMap(selectedResourcesMap)
      }
    }
    try {
      const updated = await updateResourceGroup(dataToSubmit)
      if (updated) {
        showSuccess(getString('resourceGroup.updateSuccess'))
        refetch()
      }
    } /* istanbul ignore next */ catch (e) {
      showError(e.data?.message || e.message)
    }
  }

  const onResourceSelectionChange = (resourceType: ResourceType, isAdd: boolean, identifiers?: string[]): void => {
    setIsUpdated(true)
    if (identifiers) {
      if (isAdd) {
        setSelectedResourceMap(
          produce(selectedResourcesMap, draft => {
            draft.set(resourceType, identifiers)
          })
        )
      } else {
        setSelectedResourceMap(
          produce(selectedResourcesMap, draft => {
            const resources = draft.get(resourceType)
            if (resources && typeof resources === 'object')
              draft.set(
                resourceType,
                resources.filter(el => !identifiers.includes(el))
              )
            if (draft.get(resourceType)?.length === 0) draft.delete(resourceType)
          })
        )
      }
    } else {
      if (isAdd)
        setSelectedResourceMap(
          produce(selectedResourcesMap, draft => {
            draft.set(resourceType, RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
          })
        )
      else
        setSelectedResourceMap(
          produce(selectedResourcesMap, draft => {
            draft.delete(resourceType)
          })
        )
    }
  }

  const onDragCategory = (types: ResourceType[]): void => {
    setIsUpdated(true)
    setSelectedResourceMap(
      produce(selectedResourcesMap, draft => {
        types.map(resourceType => {
          draft.set(resourceType, RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
        })
      })
    )
  }

  if (errorInGettingResourceGroup) {
    return <Page.Error message={errorInGettingResourceGroup?.message} />
  }

  const resourceGroup = resourceGroupDetails?.data?.resourceGroup
  const isHarnessManaged = resourceGroupDetails?.data?.harnessManaged

  if (!resourceGroup)
    return <Page.NoDataCard icon="resources-icon" message={getString('resourceGroup.noResourceGroupFound')} />

  return (
    <>
      <Page.Header
        size="xlarge"
        className={css.headerContainer}
        title={
          <Layout.Vertical spacing="small">
            <Breadcrumbs
              links={[
                {
                  url: routes.toAccessControl({ accountId, orgIdentifier, projectIdentifier }),
                  label: getString('accessControl')
                },
                {
                  url: routes.toResourceGroups({ accountId, orgIdentifier, projectIdentifier }),
                  label: getString('resourceGroups')
                },
                {
                  url: '#',
                  label: resourceGroup.name || ''
                }
              ]}
            />
            <Layout.Horizontal spacing="medium" flex={{ justifyContent: 'flex-start', alignItems: 'center' }}>
              <div style={{ backgroundColor: resourceGroup.color }} className={css.colorDiv}></div>
              <Layout.Vertical padding={{ left: 'medium' }} spacing="xsmall">
                <Text font="medium" lineClamp={1} color={Color.BLACK} width={700}>
                  {resourceGroup.name}
                </Text>
                <Text lineClamp={1} color={Color.GREY_400} width={700}>
                  {resourceGroup.description}
                </Text>
              </Layout.Vertical>
            </Layout.Horizontal>
          </Layout.Vertical>
        }
        toolbar={
          <Layout.Horizontal flex>
            <Layout.Vertical
              padding={{ right: 'small' }}
              border={{ right: true, color: Color.GREY_300 }}
              spacing="xsmall"
            >
              <Text>{getString('created')}</Text>
              <Text lineClamp={1} color={Color.BLACK}>
                {isHarnessManaged ? (
                  <Text lineClamp={1} color={Color.BLACK}>
                    {getString('resourceGroup.builtInResourceGroup')}
                  </Text>
                ) : (
                  <ReactTimeago date={resourceGroupDetails?.data?.createdAt || ''} />
                )}
              </Text>
            </Layout.Vertical>
            <Layout.Vertical spacing="xsmall" padding={{ left: 'small' }}>
              <Text>{getString('lastUpdated')}</Text>
              <Text lineClamp={1} color={Color.BLACK}>
                {isHarnessManaged ? (
                  <Text lineClamp={1} color={Color.BLACK}>
                    {getString('resourceGroup.builtInResourceGroup')}
                  </Text>
                ) : (
                  <ReactTimeago date={resourceGroupDetails?.data?.lastModifiedAt || ''} />
                )}
              </Text>
            </Layout.Vertical>
          </Layout.Horizontal>
        }
      />
      <Page.Body loading={updating || loading}>
        <div className={css.pageContainer}>
          <Container
            padding="xlarge"
            className={css.resourceTypeListContainer}
            border={{ right: true, color: Color.GREY_250 }}
          >
            <ResourceTypeList
              resourceCategoryMap={resourceCategoryMap}
              onResourceSelectionChange={onResourceSelectionChange}
              preSelectedResourceList={Array.from(selectedResourcesMap.keys())}
              disableAddingResources={isHarnessManaged}
            />
          </Container>
          <Container
            padding="xlarge"
            onDrop={event => {
              const resourceCategory: ResourceType | ResourceCategory = event.dataTransfer.getData('text/plain') as
                | ResourceType
                | ResourceCategory
              const types = resourceCategoryMap?.get(resourceCategory)
              if (types) {
                onDragCategory(types)
              } else onResourceSelectionChange(resourceCategory as ResourceType, true)

              event.preventDefault()
              event.stopPropagation()
            }}
            onDragOver={event => {
              event.dataTransfer.dropEffect = 'copy'
              event.preventDefault()
            }}
          >
            {!isHarnessManaged && (
              <Layout.Vertical flex={{ alignItems: 'flex-end' }} padding="medium">
                <Layout.Horizontal flex={{ justifyContent: 'flex-end' }} spacing="small">
                  {isUpdated && <Text color={Color.BLACK}>{getString('unsavedChanges')}</Text>}
                  <Button
                    text={getString('applyChanges')}
                    onClick={() => updateResourceGroupData(resourceGroup)}
                    disabled={updating || !isUpdated}
                    intent="primary"
                  />
                </Layout.Horizontal>
              </Layout.Vertical>
            )}
            <Layout.Vertical spacing="small" height="100%">
              {Array.from(selectedResourcesMap.keys()).length === 0 && (
                <Page.NoDataCard
                  message={getString('resourceGroup.dragAndDropData')}
                  icon="drag-handle-horizontal"
                  iconSize={100}
                ></Page.NoDataCard>
              )}
              {Array.from(selectedResourcesMap.keys()).length !== 0 &&
                Array.from(selectedResourcesMap.keys()).map(resourceType => {
                  const resourceValues = selectedResourcesMap.get(resourceType)
                  return (
                    resourceValues && (
                      <ResourcesCard
                        key={resourceType}
                        resourceType={resourceType}
                        resourceValues={resourceValues}
                        onResourceSelectionChange={onResourceSelectionChange}
                        disableAddingResources={isHarnessManaged}
                      />
                    )
                  )
                })}
            </Layout.Vertical>
          </Container>
        </div>
      </Page.Body>
    </>
  )
}
export default ResourceGroupDetails
