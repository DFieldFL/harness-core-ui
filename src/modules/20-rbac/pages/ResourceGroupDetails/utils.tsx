import { RbacResourceGroupTypes } from '@rbac/constants/utils'
import type { ResourceType } from '@rbac/interfaces/ResourceType'
import type { ResourceSelector } from 'services/cd-ng'

export const getSelectedResourcesMap = (
  resourceSelectorList?: ResourceSelector[]
): Map<ResourceType, string[] | string> => {
  const map: Map<ResourceType, string[] | string> = new Map()

  resourceSelectorList?.map(resourceSelector => {
    if (resourceSelector.type === RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
      map.set(resourceSelector.resourceType, RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
    else map.set(resourceSelector.resourceType, resourceSelector.identifiers)
  })

  return map
}

export const getResourceSelectorsfromMap = (map: Map<ResourceType, string[] | string>): ResourceSelector[] => {
  const resourceSelectors: ResourceSelector[] = []
  map.forEach((value, key) => {
    if (value === RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
      resourceSelectors.push({
        type: RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR,
        resourceType: key
      })
    else {
      resourceSelectors.push({
        type: RbacResourceGroupTypes.STATIC_RESOURCE_SELECTOR,
        resourceType: key,
        identifiers: value
      })
    }
  })
  return resourceSelectors
}
