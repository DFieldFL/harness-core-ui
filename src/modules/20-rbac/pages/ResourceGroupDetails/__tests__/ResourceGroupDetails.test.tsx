import React from 'react'
import { render, act, fireEvent, waitFor, RenderResult, queryByAttribute } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import routes from '@common/RouteDefinitions'
import { accountPathProps, resourceGroupPathProps } from '@common/utils/routeUtils'
import {
  getResourceTypeHandlerMock,
  getResourceGroupTypeHandlerMock,
  getResourceCategoryListMock
} from '@rbac/utils/RbacFactoryMockData'
import ResourceGroupDetails from '../ResourceGroupDetails'
import { resourceTypes, resourceGroupDetails, resourceGroupDetailsWithHarnessManaged } from './mock'

jest.mock('react-timeago', () => () => 'dummy date')

const updateResourceGroupDetails = jest.fn()
const getResourceGroupDetailsMock = jest.fn().mockImplementation(() => {
  return { data: resourceGroupDetails, refetch: jest.fn(), error: null, loading: false }
})
const updateResourceGroupDetailsMock = (data: any): Promise<{ status: string }> => {
  updateResourceGroupDetails(data.resourcegroup)
  return Promise.resolve({ status: 'SUCCESS' })
}
jest.mock('services/platform', () => ({
  useGetResourceTypes: jest.fn().mockImplementation(() => {
    return { data: resourceTypes, refetch: jest.fn(), error: null, loading: false }
  }),
  useUpdateResourceGroup: jest.fn().mockImplementation(() => ({ mutate: updateResourceGroupDetailsMock })),
  useGetResourceGroup: jest.fn().mockImplementation(() => {
    return getResourceGroupDetailsMock()
  })
}))

jest.mock('@rbac/factories/RbacFactory', () => ({
  getResourceTypeHandler: jest.fn().mockImplementation(resource => getResourceTypeHandlerMock(resource)),
  getResourceCategoryHandler: jest.fn().mockImplementation(resource => getResourceGroupTypeHandlerMock(resource)),
  getResourceCategoryList: jest.fn().mockImplementation(() => getResourceCategoryListMock())
}))

describe('Resource Groups Page', () => {
  let renderObj: RenderResult
  beforeEach(() => {
    renderObj = render(
      <TestWrapper
        path={routes.toResourceGroupDetails({ ...accountPathProps, ...resourceGroupPathProps })}
        pathParams={{ accountId: 'dummy', resourceGroupIdentifier: 'dummyResourceGroupIdentifier' }}
      >
        <ResourceGroupDetails />
      </TestWrapper>
    )
  })
  afterEach(() => {
    renderObj.unmount()
  })
  test('render data', async () => {
    const { container } = renderObj
    expect(container).toMatchSnapshot()
  })
  test('test projects selection and save', async () => {
    const { getByText, getAllByText, container } = renderObj
    const project = queryByAttribute('data-testid', container, 'CHECK-BOX-PROJECT')
    expect(project).toBeTruthy()
    fireEvent.click(project!)
    await waitFor(() => {
      expect(getAllByText('resourceGroup.all')[0]).toBeDefined()
    })
    act(() => {
      fireEvent.click(getByText('applyChanges'))
    })
    expect(updateResourceGroupDetails).toBeCalledWith({
      accountIdentifier: 'kmpySmUISimoRrJL6NL73w',
      orgIdentifier: null,
      projectIdentifier: null,
      identifier: 'ewrewew',
      name: 'ewrewew',
      resourceSelectors: [
        { type: 'DynamicResourceSelector', resourceType: 'ORGANIZATION' },
        { type: 'DynamicResourceSelector', resourceType: 'PROJECT' }
      ],
      tags: {},
      description: '',
      color: '#0063f7'
    })
  })
})
test('with harness managed resources', async () => {
  getResourceGroupDetailsMock.mockImplementation(() => {
    return { data: resourceGroupDetailsWithHarnessManaged, refetch: jest.fn(), error: null, loading: false }
  })
  const { queryByText } = render(
    <TestWrapper
      path={routes.toResourceGroupDetails({ ...accountPathProps, ...resourceGroupPathProps })}
      pathParams={{ accountId: 'dummy', resourceGroupIdentifier: 'dummyResourceGroupIdentifier' }}
    >
      <ResourceGroupDetails />
    </TestWrapper>
  )
  await waitFor(() => {
    expect(queryByText('All Organizations')).toBeDefined()
  })
  expect(queryByText('Apply Changes')).toBeNull()
})
