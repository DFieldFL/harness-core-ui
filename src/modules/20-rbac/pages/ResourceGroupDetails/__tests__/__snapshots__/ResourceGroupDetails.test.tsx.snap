// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resource Groups Page render data 1`] = `
<div>
  <div
    class="Layout--horizontal StyledProps--main container xlarge headerContainer StyledProps--flex StyledProps--padding-left-large StyledProps--padding-right-large"
  >
    <div
      class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
    >
      <div
        class="breadcrumbs"
      >
        <div>
          <a
            class="breadcrumb"
            href="/account/dummy/admin/access-control"
          >
            Access Control
          </a>
          <span
            class="separator"
          >
            /
          </span>
        </div>
        <div>
          <a
            class="breadcrumb"
            href="/account/dummy/admin/access-control/resource-groups"
          >
            Resource Groups
          </a>
          <span
            class="separator"
          >
            /
          </span>
        </div>
        <div>
          <span
            class="lastBreadcrumb"
          >
            ewrewew
          </span>
        </div>
      </div>
      <div
        class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-flex-start StyledProps--flex-alignItems-center"
      >
        <div
          class="colorDiv"
          style="background-color: rgb(0, 99, 247);"
        />
        <div
          class="Layout--vertical Layout--layout-spacing-xsmall StyledProps--main StyledProps--padding-left-medium"
        >
          <p
            class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--font-medium StyledProps--color StyledProps--color-black"
            style="--text-line-clamp: 1; width: 700px;"
          >
            ewrewew
          </p>
          <p
            class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--color StyledProps--color-grey400"
            style="--text-line-clamp: 1; width: 700px;"
          >
            
          </p>
        </div>
      </div>
    </div>
    <div
      class="Layout--horizontal StyledProps--main StyledProps--flex"
    >
      <div
        class="Layout--vertical Layout--layout-spacing-xsmall StyledProps--main StyledProps--padding-right-small StyledProps--border StyledProps--border-right StyledProps--border-color-grey300"
      >
        <p
          class="StyledProps--font StyledProps--main"
        >
          Created
        </p>
        <p
          class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--color StyledProps--color-black"
          style="--text-line-clamp: 1;"
        >
          dummy date
        </p>
      </div>
      <div
        class="Layout--vertical Layout--layout-spacing-xsmall StyledProps--main StyledProps--padding-left-small"
      >
        <p
          class="StyledProps--font StyledProps--main"
        >
          Last Updated
        </p>
        <p
          class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--color StyledProps--color-black"
          style="--text-line-clamp: 1;"
        >
          dummy date
        </p>
      </div>
    </div>
  </div>
  <div
    class="StyledProps--font StyledProps--main pageBody"
  >
    <div
      class="pageContainer"
    >
      <div
        class="StyledProps--font StyledProps--main resourceTypeListContainer StyledProps--padding-xlarge StyledProps--border StyledProps--border-right"
      >
        <div
          class="Layout--vertical Layout--layout-spacing-small StyledProps--main StyledProps--flex"
        >
          <div
            class="bp3-card bp3-elevation-0 Card--card resourceTypeCard"
            draggable="true"
          >
            <label
              class="bp3-control bp3-checkbox"
            >
              <input
                type="checkbox"
                value="SECRET"
              />
              <span
                class="bp3-control-indicator"
              />
              <span
                class="bp3-icon bp3-icon-lock StyledProps--main StyledProps--padding-left-medium StyledProps--padding-right-small"
                icon="lock"
              >
                <svg
                  data-icon="lock"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    lock
                  </desc>
                  <path
                    d="M13.96 7H12V3.95C12 1.77 10.21 0 8 0S4 1.77 4 3.95V7H1.96c-.55 0-.96.35-.96.9v6.91c0 .54.41 1.19.96 1.19h12c.55 0 1.04-.65 1.04-1.19V7.9c0-.55-.49-.9-1.04-.9zM6 7V3.95c0-1.09.9-1.97 2-1.97s2 .88 2 1.97V7H6z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
              Secrets
            </label>
          </div>
          <div
            class="bp3-card bp3-elevation-0 Card--card resourceTypeCard"
            draggable="true"
          >
            <label
              class="bp3-control bp3-checkbox"
            >
              <input
                type="checkbox"
                value="CONNECTOR"
              />
              <span
                class="bp3-control-indicator"
              />
              <span
                class="bp3-icon bp3-icon-lock StyledProps--main StyledProps--padding-left-medium StyledProps--padding-right-small"
                icon="lock"
              >
                <svg
                  data-icon="lock"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    lock
                  </desc>
                  <path
                    d="M13.96 7H12V3.95C12 1.77 10.21 0 8 0S4 1.77 4 3.95V7H1.96c-.55 0-.96.35-.96.9v6.91c0 .54.41 1.19.96 1.19h12c.55 0 1.04-.65 1.04-1.19V7.9c0-.55-.49-.9-1.04-.9zM6 7V3.95c0-1.09.9-1.97 2-1.97s2 .88 2 1.97V7H6z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
              Connectors
            </label>
          </div>
          <div
            class="bp3-card bp3-elevation-0 Card--card resourceTypeCard"
            draggable="true"
          >
            <label
              class="bp3-control bp3-checkbox"
            >
              <input
                type="checkbox"
                value="ORGANIZATION"
              />
              <span
                class="bp3-control-indicator"
              />
              <span
                class="bp3-icon bp3-icon-settings StyledProps--main StyledProps--padding-left-medium StyledProps--padding-right-small"
                icon="settings"
              >
                <svg
                  data-icon="settings"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    settings
                  </desc>
                  <path
                    d="M3 1c0-.55-.45-1-1-1S1 .45 1 1v3h2V1zm0 4H1c-.55 0-1 .45-1 1v2c0 .55.45 1 1 1h2c.55 0 1-.45 1-1V6c0-.55-.45-1-1-1zm12-4c0-.55-.45-1-1-1s-1 .45-1 1v2h2V1zM9 1c0-.55-.45-1-1-1S7 .45 7 1v6h2V1zM1 15c0 .55.45 1 1 1s1-.45 1-1v-5H1v5zM15 4h-2c-.55 0-1 .45-1 1v2c0 .55.45 1 1 1h2c.55 0 1-.45 1-1V5c0-.55-.45-1-1-1zm-2 11c0 .55.45 1 1 1s1-.45 1-1V9h-2v6zM9 8H7c-.55 0-1 .45-1 1v2c0 .55.45 1 1 1h2c.55 0 1-.45 1-1V9c0-.55-.45-1-1-1zm-2 7c0 .55.45 1 1 1s1-.45 1-1v-2H7v2z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
              Organizations
            </label>
          </div>
          <div
            class="bp3-card bp3-elevation-0 Card--card resourceTypeCard"
            draggable="true"
          >
            <label
              class="bp3-control bp3-checkbox"
            >
              <input
                type="checkbox"
                value="PROJECT"
              />
              <span
                class="bp3-control-indicator"
              />
              <span
                class="bp3-icon StyledProps--main StyledProps--padding-left-medium StyledProps--padding-right-small"
                data-icon="nav-project"
              >
                <svg
                  height="16"
                  viewBox="0 0 22 22"
                  width="16"
                >
                  <path
                    d="M1.1 4.583L11 0l9.9 4.583v12.834L11 22l-9.9-4.583zm9 5.171l-7.2-3.33v9.811l7.2 3.338V9.754zm9 6.481V6.424l-7.2 3.33v9.819zm-1.462-11.15L11 2.012 4.362 5.085 11 8.157z"
                    fill="#56778d"
                  />
                </svg>
              </span>
              Projects
            </label>
          </div>
        </div>
      </div>
      <div
        class="StyledProps--font StyledProps--main StyledProps--padding-xlarge"
      >
        <div
          class="Layout--vertical StyledProps--main StyledProps--flex StyledProps--flex-alignItems-flex-end StyledProps--padding-small"
        >
          <button
            class="bp3-button StyledProps--font Button--button StyledProps--main StyledProps--intent-primary"
            type="button"
          >
            <span
              class="bp3-button-text"
            >
              Apply Changes
            </span>
          </button>
        </div>
        <div
          class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
          style="height: 100%;"
        >
          <div
            class="bp3-card bp3-elevation-0 Card--card selectedResourceGroupCardDetails"
          >
            <div
              class="Layout--vertical StyledProps--main"
            >
              <div
                class="Layout--horizontal StyledProps--main resourceFields"
              >
                <p
                  class="StyledProps--font Text--lineclamp StyledProps--main"
                  style="--text-line-clamp: 1; display: flex; align-items: center;"
                >
                  <span
                    class="bp3-icon bp3-icon-settings StyledProps--main StyledProps--padding-right-medium"
                    icon="settings"
                  >
                    <svg
                      data-icon="settings"
                      height="20"
                      viewBox="0 0 20 20"
                      width="20"
                    >
                      <desc>
                        settings
                      </desc>
                      <path
                        d="M4 1c0-.55-.45-1-1-1S2 .45 2 1v5h2V1zM2 19c0 .55.45 1 1 1s1-.45 1-1v-6H2v6zm9-18c0-.55-.45-1-1-1S9 .45 9 1v8h2V1zm7 0c0-.55-.45-1-1-1s-1 .45-1 1v3h2V1zM9 19c0 .55.45 1 1 1s1-.45 1-1v-3H9v3zm9-14h-2c-.55 0-1 .45-1 1v3c0 .55.45 1 1 1h2c.55 0 1-.45 1-1V6c0-.55-.45-1-1-1zm-2 14c0 .55.45 1 1 1s1-.45 1-1v-8h-2v8zM4 7H2c-.55 0-1 .45-1 1v3c0 .55.45 1 1 1h2c.55 0 1-.45 1-1V8c0-.55-.45-1-1-1zm7 3H9c-.55 0-1 .45-1 1v3c0 .55.45 1 1 1h2c.55 0 1-.45 1-1v-3c0-.55-.45-1-1-1z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                  Organizations
                </p>
                <label
                  class="bp3-control bp3-radio radioBtn"
                >
                  <input
                    checked=""
                    type="radio"
                  />
                  <span
                    class="bp3-control-indicator"
                  />
                  All Organizations
                </label>
                <p
                  class="StyledProps--font Text--lineclamp StyledProps--main limitAccessCell StyledProps--color StyledProps--color-grey400"
                  style="--text-line-clamp: 1;"
                >
                  To limit access to specific set of Organizations add to this list
                </p>
                <button
                  class="bp3-button bp3-minimal StyledProps--font Button--button StyledProps--main addResourceBtn StyledProps--intent-primary"
                  type="button"
                >
                  <span
                    class="bp3-button-text"
                  >
                    + Add Organizations
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
